pkgs/development/java-modules/maven-minimal.nix

```diff
@@ -0,0 +1,147 @@
+{ stdenv, pkgs, maven }:
+
+with stdenv.lib;
+with pkgs.javaPackages;
+
+let
+  collections = import ./collections.nix { inherit pkgs; };
+  fetchMaven = pkgs.callPackage ./m2install.nix { };
+  plugins = import ./mavenPlugins.nix { inherit stdenv pkgs maven; };
+  poms = import ./poms.nix { inherit fetchMaven; };
+in rec {
+  # Maven needs all of these to function
+  mavenMinimal = flatten
+    collections.mavenLibs_2_0_6
+    ++ collections.mavenLibs_2_0_9
+    ++ collections.mavenLibs_2_2_1
+    ++ [
+      classworlds_1_1_alpha2
+      classworlds_1_1
+      commonsCli_1_0
+      commonsLang_2_1
+      commonsLang3_3_1
+      commonsLoggingApi_1_1
+      findbugsJsr305_2_0_1
+      googleCollections_1_0
+      junit_3_8_1
```

Comment:
strange, things like this could leads to https://en.wikipedia.org/wiki/Java_Classloader#JAR_hell and there are tools to work with it, like http://docs.jboss.org/tattletale/userguide/1.1/html/reports.html#multiplejarfiles
But it could more an issue for maven itself than this derivation.
How did you generate the list of dependencies?